coordinator/src/escrow_coordinator/mod.rs

```diff
@@ -134,4 +134,17 @@ impl EscrowCoordinator {
         let trade_hash: [u8; 32] = hasher.finalize().into();
         Ok((trade_hash, contract))
     }
+
+    pub async fn restart_nostr_client(
+        mut self,
+        keys: Keys,
+        relays: Vec<String>,
+    ) -> anyhow::Result<Self> {
+        warn!("Restarting nostr client...");
+        self.nostr_client.client.shutdown().await?;
```

Comment:
I wonder whether it is enough just to call `disconnect()` and after the timeout `connect()`. This would be simpler and likely more efficient.

Reply:
I agree this should be enough, the only difference is that ```shutdown()``` completely destructs the relay channels/actors, but problems probably arise from the connection, not faulty nostr-sdk state so just reconnecting should work too.